FPEngine.cc

```diff
@@ -479,6 +490,25 @@ void FPNetworkControl::probe_transmission_handler(nsock_pool nsp, nsock_event ns
         myprobe->host->fail_one_probe();
         gh_perror("Unable to send packet in %s", __func__);
       }
+      /* If Packet was sent successfully and Decoys are present */
+      else if (o.numdecoys > 1) {  
+        FPPacket *temp = new FPPacket();
+        std::map<std::string, IPv6Header*>::iterator it;
+        char addr[INET6_ADDRSTRLEN];
+        for (int decoy = 0; decoy < o.numdecoys; decoy++) {
+          if (decoy == o.decoyturn)
+            continue;
```

Comment:
This isn't exactly how `decoyturn` is supposed to work. Instead, `decoyturn` marks the index in the list of decoys where our real source address goes. Probe-sending functions generally loop over the list of decoys, sending a probe for each one. If the current index is `decoyturn`, then they also do appropriate accounting like checking the return value of the send operation, setting the time sent, etc. This is why I think you could either pass an array of `FPPacket` or `FPProbe` to `scheduleProbe`  or have some function to swap out the header (the beginning of a linked-list of `PacketElement` objects with the new address info for each decoy.